snap/info.go

```diff
@@ -1585,6 +1585,32 @@ func ReadCurrentInfo(snapName string) (*Info, error) {
 	return ReadInfo(snapName, &SideInfo{Revision: revision})
 }
 
+func ReadCurrentComponentInfo(component string, info *Info, container func(string) (Container, error)) (*ComponentInfo, error) {
+	link := filepath.Join(ComponentsBaseDir(info.InstanceName()), info.Revision.String(), component)
```

Comment:
That function takes a `ContainerPlaceInfo`, which will be a `componentPlaceInfo`. That type's constructor expects a revision, which we don't have here.

`snap.ComponentLinkPath()` could just take a the component's name, which is the only thing that is used here:

```go
func ComponentLinkPath(cpi ContainerPlaceInfo, snapRev Revision) string {
	instanceName, compName, _ := strings.Cut(cpi.ContainerName(), "+")
	compBase := ComponentsBaseDir(instanceName)
	return filepath.Join(compBase, snapRev.String(), compName)
}
```

You fine with me making that change, so that I can use the function here?

Reply:
Sure, that is fine